pkg/operator/staticresourcecontroller/static_resource_controller.go

```diff
@@ -262,6 +275,30 @@ func (c *StaticResourceController) Sync(ctx context.Context, syncContext factory
 		return nil
 	}
 
+	ready, err := c.precondition(ctx)
+	if err != nil {
+		if _, _, updateErr := v1helpers.UpdateStatus(ctx, c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:    fmt.Sprintf("%sDegraded", c.name),
+			Status:  operatorv1.ConditionTrue,
+			Reason:  "ErrorCheckingPrecondition",
+			Message: err.Error(),
+		})); updateErr != nil {
+			return updateErr
+		}
+		return err
+	}
+	if !ready {
+		if _, _, updateErr := v1helpers.UpdateStatus(ctx, c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:    fmt.Sprintf("%sDegraded", c.name),
```

Comment:
I would not expect an operator to go degraded because of some precondition. A precondition elsewhere is something that disabled or delays a controllers to run, but has no visibility to outside.

Reply:
I checked what other controllers are doing and it seems the `workload.go` reports degraded as well. Here's the relevant part:

```
	if !preconditionsReady {
		var message string
		for _, err := range errs {
			message = message + err.Error() + "\n"
		}
		if len(message) == 0 {
			message = "the operator didn't specify what preconditions are missing"
		}

		// we are degraded, not available and we are not progressing

		deploymentDegradedCondition.Status = operatorv1.ConditionTrue
		deploymentDegradedCondition.Reason = "PreconditionNotFulfilled"
		deploymentDegradedCondition.Message = message

		deploymentAvailableCondition.Status = operatorv1.ConditionFalse
		deploymentAvailableCondition.Reason = "PreconditionNotFulfilled"

		deploymentProgressingCondition.Status = operatorv1.ConditionFalse
		deploymentProgressingCondition.Reason = "PreconditionNotFulfilled"

		return kerrors.NewAggregate(errs)
	}
```

Source: https://github.com/openshift/library-go/blob/a467dafcef6988afbcce43400c339d7ca68a3298/pkg/operator/apiserver/controller/workload/workload.go#L196-L218

So for symmetry reasons, I'd suggest to leave this as is.